codice/ddf
# DDF-2048: Provides support for ingesting files through an FTP Endpoint

#### What does this PR do?
This PR provides an FTP endpoint for ingesting files into the DDF catalog. Features:
-Basic Auth
-PUT & MPUT operations

#### Who is reviewing it (please choose AT LEAST two reviewers that need to approve the PR before it can get merged; if a component team is listed, at least one of its members needs to approve)?
@mweser @troymohl @dcruver @ricklarsen 
#### Choose 2 committers to review/merge the PR (please choose ONLY two committers from below, delete the rest).
@jaymcnallie
@kcwire
#### How should this be tested?
-Install the FTP feature in DDF Catalog -> Features
-Connect to the FTP server using your favorite client. Address is `localhost` and default port is `8021`. The login is the /etc/users.properties users. Default is admin/admin
-Ingest a file with the PUT or MPUT commands and verify it is correctly ingested and searchable in the Search UI.
_NOTE:_ Ingesting a .ntf file without first installing the Alliance Imagery App will cause Tika to throw an error and cause the FTP server to hang. There is a bug in Tika which is fixed in 1.12. DDF-2127 has been created to address this.
#### Any background context you want to provide?
The purpose of the ticket is to directly ingest a file without first temporarily writing the file to the file system, thus avoiding extra IO overhead.
#### What are the relevant tickets?
[DDF-2048](https://codice.atlassian.net/browse/DDF-2048)
#### Screenshots (if appropriate)
N/A
#### Checklist:
